piptools/resolver.py

```diff
@@ -87,14 +88,17 @@ def resolve(self, max_rounds=10):
 
     def _check_constraints(self):
         for constraint in chain(self.our_constraints, self.their_constraints):
-            if constraint.link is not None and not constraint.editable:
-                msg = ('pip-compile does not support URLs as packages, unless they are editable '
-                       '(perhaps add -e option?)')
-                raise UnsupportedConstraint(msg)
+            if constraint.link and constraint.link.scheme not in vcs.all_schemes:
```

Comment:
Yes, that would be `Link.is_artifact` (see desc for https://github.com/blueyed/pip-tools/commit/b6c4a0b12fcc5f5359dcd72b82d7dd3f54a5e20f).